### PR TITLE
docs(cn): translate reference/rules/index into Chinese

### PR DESCRIPTION
**Pull request by Yucohny**
翻译 reference/rules/index。

### CODE REVIEW

File: src/content/reference/rules/index.md [hunk 1]
```diff
@@ -1,52 +1,54 @@
...
 
-[Purity in Components and Hooks](/reference/rules/components-and-hooks-must-be-pure) is a key rule of React that makes your app predictable, easy to debug, and allows React to automatically optimize your code.
+[组件与 Hook 必须是纯粹的](/reference/rules/components-and-hooks-must-be-pure) 是 React 的一个关键规则，它使应用程序可预测，易于调试，并允许 React 自动优化代码。
+
+* [组件必须是幂等的](/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent) —— React 组件假定始终返回与其输入（props、state 与上下文）相同的输出。
+* [副作用必须在渲染外运行](/reference/rules/components-and-hooks-must-be-pure#side-effects-must-run-outside-of-render) —— Effect 不应在渲染中运行，因为 React 可能多次渲染组件以创建最佳的用户体验。
+* [Props 和 state 是不可变的](/reference/rules/components-and-hooks-must-be-pure#props-and-state-are-immutable) —— 组件的 props 和 state 是针对单次渲染的不可变快照。永远不要直接改变它们。
+* [返回给 Hook 的值和参数是不可变的](/reference/rules/components-and-hooks-must-be-pure#return-values-and-arguments-to-hooks-are-immutable) —— 一旦值被传递给一个 Hook，那么就不应该修改它们。与 JSX 中的 props 类似，传递给 Hook 的值在传递后变为不可变。
+* [值在传递给 JSX 后是不可变的](/reference/rules/components-and-hooks-must-be-pure#values-are-immutable-after-being-passed-to-jsx) —— 不要改变已经在 JSX 中使用的值，应当在创建 JSX 之前改变。
```
Comment (awxiaoxian2020) [inline src/content/reference/rules/index.md:34]:
```suggestion
```

**Review by awxiaoxian2020 — CHANGES_REQUESTED** [review]

File: src/content/reference/rules/index.md [hunk 2]
```diff
@@ -1,52 +1,53 @@
 ---
-title: Rules of React
+title: React 规则
 ---
 
 <Intro>
-Just as different programming languages have their own ways of expressing concepts, React has its own idioms — or rules — for how to express patterns in a way that is easy to understand and yields high-quality applications.
+正如不同的编程语言有其表达概念的方式一样，React 也有自己的俗语与规则，它们以一种易于理解并能帮助实现高质量应用程序的方式表达出来
```
Comment (awxiaoxian2020) [inline src/content/reference/rules/index.md:6]:
```suggestion
正如不同的编程语言有其表达概念的方式一样，React 也有自己的俗语与规则，它们以一种易于理解并能帮助实现高质量应用程序的方式表达出来。
```

**Review by awxiaoxian2020 — CHANGES_REQUESTED** [review]

File: src/content/reference/rules/index.md [hunk 3]
```diff
@@ -1,52 +1,51 @@
...
 
 ---
 
 <Note>
-To learn more about expressing UIs with React, we recommend reading [Thinking in React](/learn/thinking-in-react).
+要了解更多有关使用 React 表达 UI 的信息，我们建议阅读 [React 哲学](/learn/thinking-in-react)。
 </Note>
 
-This section describes the rules you need to follow to write idiomatic React code. Writing idiomatic React code can help you write well organized, safe, and composable applications. These properties make your app more resilient to changes and makes it easier to work with other developers, libraries, and tools.
+本节描述需要遵循的规则，以编写俗语化的 React 代码。编写俗语化的 React 代码可以帮助编写组织良好、安全、可组合的应用程序。这些特性使应用程序更具抗变性，并且更易于与其他开发人员、库和工具一起使用。
```
Comment (awxiaoxian2020) [inline src/content/reference/rules/index.md:17]:
```suggestion
本节描述需要遵循的规则，以编写地道的 React 代码。编写俗语化的 React 代码可以帮助编写组织良好、安全、可组合的应用程序。这些特性使应用程序更具抗变性，并且更易于与其他开发人员、库和工具一起使用。
```

File: src/content/reference/rules/index.md [hunk 4]
```diff
@@ -1,52 +1,51 @@
 ---
-title: Rules of React
+title: React 规则
 ---
 
 <Intro>
-Just as different programming languages have their own ways of expressing concepts, React has its own idioms — or rules — for how to express patterns in a way that is easy to understand and yields high-quality applications.
+正如不同的编程语言有其表达概念的方式一样，React 也有自己的俗语与规则，它们以一种易于理解并能帮助实现高质量应用程序的方式表达出来。
```
Comment (awxiaoxian2020) [inline src/content/reference/rules/index.md:6]:
```suggestion
正如不同的编程语言有其表达概念的方式一样，React 也有自己的习惯与规则，它们以一种易于理解并能帮助实现高质量应用程序的方式表达出来。
```

**Review by awxiaoxian2020 — CHANGES_REQUESTED** [review]
ping @Huxpro to review this :)

**Comment by Huxpro** [comment]
> ping @Huxpro to review this :)

I just noticed this! Why did @Yucohny close it?

**Pull request closed without merging**